vault/resource_kubernetes_auth_backend_role.go

```diff
@@ -0,0 +1,338 @@
+
+	iBoundServiceAccountNamespaces := resp.Data["bound_service_account_namespaces"].([]interface{})
+	boundServiceAccountNamespaces := make([]string, 0, len(iBoundServiceAccountNamespaces))
+
+	for _, iBoundServiceAccountNamespace := range iBoundServiceAccountNamespaces {
+		boundServiceAccountNamespaces = append(boundServiceAccountNamespaces, iBoundServiceAccountNamespace.(string))
+	}
+
+	d.Set("bound_service_account_namespaces", boundServiceAccountNamespaces)
+
+	iPolicies := resp.Data["policies"].([]interface{})
+	policies := make([]string, 0, len(iPolicies))
+
+	for _, iPolicy := range iPolicies {
+		policies = append(policies, iPolicy.(string))
+	}
+
+	d.Set("policies", policies)
+
+	ttl, err := resp.Data["ttl"].(json.Number).Int64()
+	if err != nil {
+		return fmt.Errorf("expected `ttl` %q to be a number, isn't", resp.Data["ttl"])
+	}
+	d.Set("ttl", ttl)
+
+	maxTTL, err := resp.Data["max_ttl"].(json.Number).Int64()
+	if err != nil {
+		return fmt.Errorf("expected `max_ttl` %q to be a number, isn't", resp.Data["max_ttl"])
+	}
+	d.Set("max_ttl", maxTTL)
+
+	period, err := resp.Data["period"].(json.Number).Int64()
+	if err != nil {
+		return fmt.Errorf("expected `period` %q to be a number, isn't", resp.Data["period"])
+	}
+	d.Set("period", period)
+
+	return nil
+}
+
+func kubernetesAuthBackendRoleUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	path := d.Id()
+
+	log.Printf("[DEBUG] Updating Kubernetes auth backend role %q", path)
+
+	iBoundServiceAccountNames := d.Get("bound_service_account_names").([]interface{})
+	boundServiceAccountNames := make([]string, 0, len(iBoundServiceAccountNames))
+	for _, iBoundServiceAccountName := range iBoundServiceAccountNames {
+		boundServiceAccountNames = append(boundServiceAccountNames, iBoundServiceAccountName.(string))
+	}
+
+	iBoundServiceAccountNamespaces := d.Get("bound_service_account_namespaces").([]interface{})
+	boundServiceAccountNamespaces := make([]string, 0, len(iBoundServiceAccountNamespaces))
+	for _, iBoundServiceAccountNamespace := range iBoundServiceAccountNamespaces {
+		boundServiceAccountNamespaces = append(boundServiceAccountNamespaces, iBoundServiceAccountNamespace.(string))
+	}
+
+	iPolicies := d.Get("policies").([]interface{})
+	policies := make([]string, 0, len(iPolicies))
+	for _, iPolicy := range iPolicies {
+		policies = append(policies, iPolicy.(string))
+	}
+
+	data := map[string]interface{}{
+		"bound_service_account_names":      strings.Join(boundServiceAccountNames, ","),
+		"bound_service_account_namespaces": strings.Join(boundServiceAccountNamespaces, ","),
+		"policies":                         strings.Join(policies, ","),
+	}
+
+	if v, ok := d.GetOk("ttl"); ok {
+		data["ttl"] = v.(int)
+	}
+
+	if v, ok := d.GetOk("max_ttl"); ok {
+		data["max_ttl"] = v.(int)
+	}
+
+	if v, ok := d.GetOk("period"); ok {
+		data["period"] = v.(int)
+	}
+
+	_, err := client.Logical().Write(path, data)
+
+	d.SetId(path)
```

Comment:
Could this be moved up too?

Reply:
Ditto as https://github.com/terraform-providers/terraform-provider-vault/pull/94#discussion_r220737061